compiler/damlc/daml-compiler/src/DA/Daml/Compiler/Repl.hs

```diff
@@ -373,10 +375,24 @@ runRepl
     -> IO ()
 runRepl importPkgs opts replClient logger ideState = do
     imports <- loadPackages importPkgs replClient ideState
+    -- Typecheck ones to get the GlobalRdrEnv
+    let initialLineNumber = 0
+    dflags <- liftIO $
+         hsc_dflags . hscEnv <$>
+         runAction ideState (use_ GhcSession $ lineFilePath initialLineNumber)
+    (_, tcr) <-
+        (runExceptT $ printDelayedDiagnostics $ tryTypecheck initialLineNumber $
+         T.pack (unlines $ moduleHeader dflags (importFromList imports) initialLineNumber)
```

Comment:
Maybe factor this part out? There are another couple instances of this in the module.

Reply:
good idea, done!